### ISSUE TITLE
Context URL of single-valued containment navigation property

### ISSUE THREAD

**Issue by ralfhandl** [oasis-tcs/odata-specs#2000]
Section [10.3 Entity](https://docs.oasis-open.org/odata/odata/v4.02/csd01/part1-protocol/odata-v4.02-csd01-part1-protocol.html#Entity) states that

> Context URL template:
>    `{context-url}#{entity-set}/$entity`

and

> If the entity is contained, then entity-set is the top-level entity set or singleton followed by the path to the containment navigation property of the containing entity.

Is the suffix `/$entity` only relevant for collection-valued containment, or also for single-valued containment?

Clarify and add single-valued example.

**Comment by HeikoTheissen** [comment]
> Is the suffix `/$entity` only relevant for collection-valued containment, or also for single-valued containment?

Also for single-valued containment. The suffix is omitted for singletons, but these are always top-level in the container. A single-valued containment navigation property contains a single entity, not a singleton.

(By contrast, a non-containment navigation property can target a singleton through a suitable navigation property binding.)

**Comment by ralfhandl** [comment]
I disagree: a singleton is basically a single-valued containment navigation of the entity container.

They behave the same in all other aspects, why should their context URL behave differently?

**Comment by HeikoTheissen** [comment]
> They behave the same in all other aspects, why should their context URL behave differently?

Because [OData-CSDL, section 13.3] says:
> Singletons are top-level single-valued resources.

Perhaps change that?

**Comment by ralfhandl** [comment]
That statement is correct. And single-valued containment navigation properties are nested single-valued resources.

So rather clarify the section on context URL and use the service-relative path of the single-valued resource as its context URL, consistently without the suffix `/$entity`.

**Comment by ralfhandl** [comment]
@mikepizzo and @Hubert-Heijkers would expect same treatment as singletons, without suffix `/$entity`

**Comment by ralfhandl** [comment]
Terminology
* canonical-collection
* canonical-singleton
* canonical-member
